app/src/main/java/agersant/polaris/features/queue/QueueFragment.java

```diff
@@ -0,0 +1,236 @@
+                    return;
+                }
+                switch (intent.getAction()) {
+                    case PlaybackQueue.REMOVED_ITEM:
+                    case PlaybackQueue.REMOVED_ITEMS:
+                        updateTutorial();
+                        break;
+                    case PlaybackQueue.QUEUED_ITEMS:
+                    case PlaybackQueue.OVERWROTE_QUEUE:
+                        adapter.notifyDataSetChanged();
+                        updateTutorial();
+                        break;
+                    case PolarisPlayer.OPENING_TRACK:
+                    case PolarisPlayer.PLAYING_TRACK:
+                    case OfflineCache.AUDIO_CACHED:
+                    case OfflineCache.AUDIO_REMOVED_FROM_CACHE:
+                    case DownloadQueue.WORKLOAD_CHANGED:
+                        adapter.notifyItemRangeChanged(0, adapter.getItemCount());
+                        break;
+                }
+            }
+        };
+        requireActivity().registerReceiver(receiver, filter);
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        setHasOptionsMenu(true);
+
+        PolarisState state = PolarisApplication.getState();
+        playbackQueue = state.playbackQueue;
+        player = state.player;
+        offlineCache = state.offlineCache;
+        downloadQueue = state.downloadQueue;
+
+        FragmentQueueBinding binding = FragmentQueueBinding.inflate(inflater);
+
+        recyclerView = binding.queueRecyclerView;
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new LinearLayoutManager(requireContext()));
+        DefaultItemAnimator animator = new DefaultItemAnimator() {
+            @Override
+            public boolean animateRemove(RecyclerView.ViewHolder holder) {
+                holder.itemView.setAlpha(0.f);
+                return false;
+            }
+
+            @Override
+            public boolean canReuseUpdatedViewHolder(@NonNull RecyclerView.ViewHolder viewHolder) {
+                return true;
+            }
+        };
+        recyclerView.setItemAnimator(animator);
+
+        tutorial = binding.queueTutorial;
+
+        populate();
+        updateTutorial();
+
+        return binding.getRoot();
+    }
+
+    private void updateTutorial() {
+        boolean empty = adapter.getItemCount() == 0;
+        if (empty) {
+            tutorial.setVisibility(View.VISIBLE);
+        } else {
+            tutorial.setVisibility(View.GONE);
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        subscribeToEvents();
+        updateTutorial();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        requireActivity().unregisterReceiver(receiver);
+        receiver = null;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        adapter.notifyDataSetChanged();
```

Comment:
This exact line is probably not the culprit, but I found a minor bug in this space.

When navigating to the `Queue`, the SD card icons indicating offline items blink `on -> off -> on`, which looks a bit jank. Before this change, the queue would always open with no icons and then show them a fraction of second later (`off -> on`).

GIF capture:

![blink](https://user-images.githubusercontent.com/817256/103161318-c02f7680-4794-11eb-8725-08d49ae6669d.gif)

Reply:
The line was the problem indeed. By moving the `notifiyDataSetChanged()` into `onStart()` the flashing went away. It was redundant in `onResume()` since the receiver is only unregistered in `onStop()` not in `onPause()`.